src/views/project/common/InstitutionSearchInput.vue

```diff
@@ -0,0 +1,64 @@
+<script setup lang="ts">
+import axios from 'axios'
+import SearchSelectInput from '@/components/project/SearchSelectInput.vue'
+import { useProjectInstitutionStore } from '@/stores/ProjectsInstitutionStore'
+
+const props = defineProps<{
+  projectId: number
+}>()
+
+const emit = defineEmits(['updateParent'])
+const ProjectsInstitutionStore = useProjectInstitutionStore()
+
+async function searchInstitutions(searchTerm: string) {
+  try {
+    const url = `${import.meta.env.VITE_API_URL}/projects/${
+      props.projectId
+    }/institutions/search`
+
+    const response = await axios.get(url, {
+      params: { searchTerm: searchTerm },
+    })
+
+    return response.data
+  } catch (e) {
+    console.error('Error getting Institutions\n', e)
+  }
+}
+
+function getInstitutionId(institution: any) {
+  return institution.institution_id
+}
+
+function getInstitutionName(institution: any) {
+  return institution.name
+}
+
+async function getInstitution(institutionId: number) {
+  return await ProjectsInstitutionStore.getInstitutionById(
+    props.projectId,
+    institutionId
+  )
```

Comment:
This endpoint is not needed. This is used because the `institutionId` is not within the `ProjectsInstitutionStore` and you need to return the institution id and name.

However, you already fetched the institutions with the `searchInstitutions` rather than request the institution again. You can cache the results and use the cache here to return the institution.

Namely, you can create: `const institutions = new Map()` and insert values in `searchInstitutions`